boards: ambiq: apollo4p_evb: fix multiple CI failure

aaronyegx opened the pull request:
This PR is trying to fix multiple CI failures which were introduced from https://github.com/zephyrproject-rtos/zephyr/pull/70842.
Adds the missing flash controller node in apollo4p_evb to address the CI failure https://github.com/zephyrproject-rtos/zephyr/issues/74212. 
Remove the untestable cache kconfig from apollo4x boards to address the CI failure https://github.com/zephyrproject-rtos/zephyr/issues/72775, https://github.com/zephyrproject-rtos/zephyr/issues/73443 at this moment.
And enable the twister for apollo4x platforms. Refer to https://github.com/zephyrproject-rtos/zephyr/pull/74232.

Sorry for inconvenience.

Fixes #74212
Fixes #72775
Fixes #73443

aaronyegx commented:
It seems the ambiq_hal has enabled the ICACHE_PRESENT and DCACHE_PRESENT for Apollo4x SoC. Disable the cache kconfig is working for the CI failure addressing. Need some time to work out the solution.

RichardSWheatley commented on soc/ambiq/apollo4x/Kconfig:
```diff
@@ -11,6 +11,4 @@ config SOC_SERIES_APOLLO4X
 	select CPU_HAS_ARM_MPU
 	select HAS_SWO
 	select AMBIQ_HAL
-	select CPU_HAS_DCACHE
```
You will get errors as these are enabled in the HAL.

aaronyegx replied:
I know, as I mentioned in last comment "Disable the cache kconfig is not working". And I mark the PR to "Draft" and add "DNM" label. Please try to address https://github.com/zephyrproject-rtos/zephyr/issues/72775, https://github.com/zephyrproject-rtos/zephyr/issues/73443 if time available, or we can disable the cache in both HAL and SoC kconfig.

RichardSWheatley replied:
If they are disabling the twister test already, is there also a need to change the cake and fig and reenable the test? I don't think that makes much sense.

aaronyegx replied:
Disabling the twister test for Apollo4x is just a temporary solution to unblock others' PRs and the coming 3.7.0 release. We need the twister to check our changes for Apollo4x in our next PRs. It is reporting "apollo4p_evb" is unrecognized platform when we push changes of apollo4p_evb then checks failure.

RichardSWheatley commented:
@aaronyegx if this passes the twister test, post here and I will approve.

RichardSWheatley reviewed (approved):
Thanks Aaron!

swift-tk commented:
This PR should go in after https://github.com/zephyrproject-rtos/zephyr/pull/71769 as it advances hal_ambiq commit id in west.yaml.

aaronyegx committed:
boards: ambiq: apollo4p_evb: add missing flash node

This commit adds the missing flash controller node in apollo4p_evb.

Signed-off-by: Aaron Ye <aye@ambiq.com>

aaronyegx committed:
soc: ambiq: apollo4x: remove untestable cache kconfig

This commit removes the untestable data and instruction cache kconfig
which causes CI failure.

Signed-off-by: Aaron Ye <aye@ambiq.com>

aaronyegx committed:
boards: ambiq: enable twister on apollo4x evb

This commit enables the twister on apollo4p_evb and apollo4p_blue_kxr_evb.

Signed-off-by: Aaron Ye <aye@ambiq.com>

aaronyegx commented:
Rebase to resolve the conflict.

aescolar commented:
@RichardSWheatley @aaronyegx is this in principle ready and fixing all the previous issues?

aescolar commented:
> @RichardSWheatley @aaronyegx is this in principle ready and fixing all the previous issues?

I answer to myself: Not really:

twister -p apollo4p_evb -T samples/drivers/flash_shell/ 

cmsis_core_m.h:56:2: error: #error "__ICACHE_PRESENT and CONFIG_CPU_HAS_ICACHE are not set to the same value"
   56 | #error "__ICACHE_PRESENT and CONFIG_CPU_HAS_ICACHE are not set to the same value"
      |  ^~~~~

aaronyegx commented:
> twister -p apollo4p_evb -T samples/drivers/flash_shell

Did you use the updated hal_ambiq commit# for testing?

aaronyegx commented:
> > @RichardSWheatley @aaronyegx is this in principle ready and fixing all the previous issues?
> 
> I answer to myself: Not really:
> 
> twister -p apollo4p_evb -T samples/drivers/flash_shell/
> 
> cmsis_core_m.h:56:2: error: #error "__ICACHE_PRESENT and CONFIG_CPU_HAS_ICACHE are not set to the same value" 56 | #error "__ICACHE_PRESENT and CONFIG_CPU_HAS_ICACHE are not set to the same value" | ^~~~~

@aescolar Did you use the updated hal_ambiq commit# for testing.

AlessandroLuo commented:
LGTM
